### PR TITLE
Trigger refactor

### LINKED ISSUE #255: do triggering with structured concurrency
Refactor how triggering is done to eliminate the sue of go channels to the extent possible. Instead split `processSegments` into two functions, add another waitgroup loop that does triggering. Then do the group triggering in a function after that waitgroup loop.

### PR DESCRIPTION
**Pull request by joefowler**
Complete redesign of the triggering logic and especially the `TriggerBroker`, from which we are removing the long-running goroutine and all channels for its input and output. The code is no shorter than before, but we hope it should be clearer. It uses concurrency only where performance requires it (spinning a goroutine for each channel's data processing) and uses the classic simple concurrency model where all _N_ concurrent processes are waited on before proceeding. Fixes #255.

### CODE REVIEW

File: group_trigger.go [hunk 1]
```diff
@@ -200,20 +190,21 @@ func (broker *TriggerBroker) Distribute(primaries map[int]triggerList) (map[int]
 	for idx, tlist := range primaries {
 		broker.latestPrimaries[idx] = tlist.frames
 		nprimaries += len(tlist.frames)
-		err := broker.triggerCounters[idx].observeTriggerList(&tlist)
+		err := broker.triggerCounters[idx].countNewTriggers(&tlist)
 		if err != nil {
 			log.Printf("triggering assumptions broken!\n%v\n%v\n%v", err,
```
Comment (ggggggggg) [inline group_trigger.go:195]: I was going to suggest changing this panic to just log, which it seems you did.... so good

Comment (joefowler) [reply group_trigger.go:196]: Don't think this was ever a panic, at least not recently.

**Comment by ggggggggg** [comment]
Is it getting send once per channel per second instead of once per second? I think it was sent at the right rate before.

**Comment by ggggggggg** [comment]
fixes #260 
fixes #259 
fixes #256 
fixes #255

File: edge_multi_test.go [hunk 2]
```diff
@@ -109,46 +60,97 @@ func TestEdgeMultiParts1(t *testing.T) {
...
+				nmonotone: 1, npre: 2, nsamp: 4},
+			[]RecordSpec{{firstRisingFrameIndex: 2, npre: 2, nsamp: 4},
+				{firstRisingFrameIndex: 4, npre: 2, nsamp: 4},
+				{firstRisingFrameIndex: 6, npre: 2, nsamp: 4},
+				{firstRisingFrameIndex: 8, npre: 2, nsamp: 4},
+				{firstRisingFrameIndex: 10, npre: 2, nsamp: 4},
+				{firstRisingFrameIndex: 12, npre: 2, nsamp: 4},
+				{firstRisingFrameIndex: 14, npre: 2, nsamp: 4},
+				{firstRisingFrameIndex: 16, npre: 2, nsamp: 4}},
+			"edgeMultiComputeAppendRecordSpecs: 2 records far enough apart to trigger both, negative going",
```
Comment (ggggggggg) [inline edge_multi_test.go:120]: this error string is not accurate

**Pull request merged**